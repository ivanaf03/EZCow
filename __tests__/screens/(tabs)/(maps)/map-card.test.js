import React from 'react';

import { render, waitFor, fireEvent } from "@testing-library/react-native";
import MapCard from "../../../../app/(tabs)/(maps)/map-card";

jest.mock("@fortawesome/react-native-fontawesome", () => ({
    FontAwesomeIcon: () => null,
}));

describe("MapCard", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", async () => {
        const tree = render(<MapCard map={{
            name: "Mapa Test",
            latitude: 10,
            longitude: 10,
        }} />);
        await waitFor (() => expect(tree).toMatchSnapshot());
    });

    it("should call onDelete with map data when pressing delete", async () => {
        const onDeleteMock = jest.fn();
        const tree = render(<MapCard map={{
            name: "Mapa Test",
            latitude: 10,
            longitude: 10,
        }} onDelete={onDeleteMock} />);
        const deleteButton = tree.getByTestId("delete-map-button");
        fireEvent.press(deleteButton);
        expect(onDeleteMock).toHaveBeenCalledTimes(1);
    });
});