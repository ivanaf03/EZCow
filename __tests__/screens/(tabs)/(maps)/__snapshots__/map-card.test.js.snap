// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapCard should render correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "marginBottom": 8,
        "marginTop": 8,
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "marginRight": 24,
      },
      {
        "backgroundColor": "#272727",
      },
      {
        "borderBottomRightRadius": 24,
        "borderTopRightRadius": 24,
      },
      {
        "borderRightWidth": 4,
      },
      {
        "borderBottomColor": "#6bc0eb",
        "borderLeftColor": "#6bc0eb",
        "borderRightColor": "#6bc0eb",
        "borderTopColor": "#6bc0eb",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "maxWidth": "50%",
          },
          {
            "overflow": "hidden",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "fontFamily": "Nunito-Bold",
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "fontFamily": "Nunito-Bold",
            },
            [
              {
                "marginLeft": 8,
              },
            ],
          ]
        }
      >
        Mapa Test
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "backgroundColor": "#272727",
          },
          {
            "borderBottomWidth": 2,
            "borderLeftWidth": 2,
            "borderRightWidth": 2,
            "borderTopWidth": 2,
          },
          {
            "borderBottomColor": "#6bc0eb",
            "borderLeftColor": "#6bc0eb",
            "borderRightColor": "#6bc0eb",
            "borderTopColor": "#6bc0eb",
          },
          {
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
          },
        ]
      }
      testID="delete-map-button"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Nunito-SemiBold",
            },
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            [
              {
                "marginLeft": 16,
              },
            ],
          ]
        }
      >
        Borrar finca
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "height": 240,
        },
        {
          "backgroundColor": "#272727",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "marginTop": 16,
        },
        [
          {
            "marginTop": 16,
          },
        ],
      ]
    }
  >
    <AIRMap
      initialRegion={null}
      mapType="satellite"
      onChange={[Function]}
      onMapReady={[Function]}
      region={
        {
          "latitude": 10,
          "latitudeDelta": 0.01,
          "longitude": 10,
          "longitudeDelta": 0.01,
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <AIRMapMarker
        coordinate={
          {
            "latitude": 10,
            "longitude": 10,
          }
        }
        onPress={[Function]}
        style={
          [
            {
              "backgroundColor": "transparent",
              "position": "absolute",
            },
            undefined,
          ]
        }
        title="Mapa Test"
      />
    </AIRMap>
  </View>
</View>
`;
